content/blog/nr-detect-sqlia/index.md

```diff
@@ -0,0 +1,287 @@
+---
+path: '/nr-detect-sqli'
+title: 'Guest Post: Detect SQL injections with Pixie'
+date: 2021-10-14T00:00:00.000+00:00
+featured_image: hero-image.png
+categories: ['Pixie Team Blogs', 'Announcements', 'Kubernetes']
+authors: ['Elaine Laguerta', 'Hannah Stepanek', 'Robert Prast - New Relic']
+emails: ['elaguerta@newrelic.com', 'hstepanek@newrelic.com', 'rprast@newrelic.com']
+featured: true
+---
+*FYI - The team will demo this SQL injection detection workflow at KubeCon! The demo will be live at KubeCon 2021 [Friday, October 15, 11 am PDT](https://kccncna2021.sched.com/event/lV4c). The source code for the demo is [available at the Pixie demo repo](https://github.com/pixie-io/pixie-demos/tree/add-sql-injection-demo/sql-injection-demo).*   
+*This is a guest post by employees of New Relic.*  
+
+---
+
+Think about the possible security vulnerabilities in code that you ship. What keeps you up at night? You’d be wise to answer “SQL injection” -- after all, since 2003 it’s been on the OWASP’s (Open Web Application Security Project) [top 10 list of CVEs](https://owasp.org/Top10/) (Common Vulnerabilities and Exposures).  
+
+Imagine you have an endpoint that takes an id query parameter:
+```
+http://example.com.com/api/host?id=1
+```
+The id parameter, it turns out, is not correctly sanitized and it is injectable. A hacker could come along and do something like: 
+```
+http://example.com.com/api/host?id=1 UNION SELECT current_user, NULL FROM user 
+```
+Which, at the database level would run the following query: 
+```sql
+SELECT name, ip FROM host WHERE id=1 UNION SELECT current_user, NULL FROM user
+```
+Instead of returning the host data that belongs to that id, the endpoint would now return the database username. As you can imagine this could be really bad as the hacker has just exposed a huge vulnerability within this endpoint. Any user using this vulnerability could get any data they wanted out of the database. 
+
+We won’t link to any news stories about devastating hacks; that’s not why we’re here. As developers and security engineers, we believe our job isn’t to scare, but to empower. Pixie is a powerful observability platform, and as security people we saw some unique opportunities to apply Pixie to security use cases. We’re here to show you how you can use Pixie to proactively detect and report SQL injection attempts, while your application is live.   
```

Comment:
Response to @nserrino 's comment: "Can we add a sentence that says something like, Pixie is an observability platform but we are security people and we wanted to see how we could apply Pixie to security use cases?"